src/parser/jobs/sch/modules/Aetherflow.tsx

```diff
@@ -0,0 +1,231 @@
+					<Table.Cell textAlign="right"><ActionLink {...this.data.actions.SCH_ENERGY_DRAIN}/></Table.Cell>
+					<Table.Cell colSpan="3">{this.totalCastsByConsumeAction.get(this.data.actions.SCH_ENERGY_DRAIN.id) ?? 0}</Table.Cell>
+				</Table.Row>
+				<Table.Row>
+				</Table.Row>
+			</Table.Body>
+		</Table>
+		<Table>
+			<Table.Header>
+				<Table.Row>
+					<Table.HeaderCell><Trans id="sch.aetherflow.cast">Cast</Trans></Table.HeaderCell>
+					<Table.HeaderCell><Trans id="sch.aetherflow.cast-time">Cast Time</Trans></Table.HeaderCell>
+					<Table.HeaderCell><Trans id="sch.aetherflow.drift">Drift</Trans></Table.HeaderCell>
+					<Table.HeaderCell><Trans id="sch.aetherflow.abilities-used">Abilities Used</Trans></Table.HeaderCell>
+					<Table.HeaderCell><Trans id="sch.aetherflow.wasted-stacks">Wasted Stacks</Trans></Table.HeaderCell>
+				</Table.Row>
+			</Table.Header>
+			<Table.Body>
+				{this.aetherflowWindows.map(aetherflowWindow => {
+					return <Table.Row key={aetherflowWindow.timestamp}>
+						<Table.Cell><ActionLink {...this.data.getAction(aetherflowWindow.aetherflowGenerateActionId)}/></Table.Cell>
+						<Table.Cell>
+							<Button
+								circular
+								compact
+								size="mini"
+								icon="time"
+								onClick={() => this.scrollToAetherflowTimeline(aetherflowWindow.timestamp)}
+							/>{this.parser.formatEpochTimestamp(aetherflowWindow.timestamp)}</Table.Cell>
+						<Table.Cell>{aetherflowWindow.drift > 0 && this.parser.formatDuration(aetherflowWindow.drift)}</Table.Cell>
+						<Table.Cell>
+							<Grid>
+								{aetherflowWindow.aetherflowConsumeActions.map((action, index) => <Grid.Column key={index} width={5}>
+									<Grid.Row>
+										<ActionLink {...this.data.getAction(action.actionId)}/>
+									</Grid.Row>
+									<Grid.Row>
+										<Button
+											circular
+											compact
+											size="mini"
+											icon="time"
+											onClick={() => this.scrollToAetherflowTimeline(aetherflowWindow.timestamp)}
+										/>
+										{this.parser.formatEpochTimestamp(action.timestamp)}
+									</Grid.Row>
+								</Grid.Column>)}
+							</Grid>
+						</Table.Cell>
+						<Table.Cell>{this.AETHERFLOW_CHARGES_PER_CAST - aetherflowWindow.aetherflowConsumeActions.length || '-'}</Table.Cell>
+					</Table.Row>
+				})}
+			</Table.Body>
+		</Table></>
+	}
+
+	private onGenerateAetherflow(event: Events['action']) {
+		// Calculate any values that depend on the previous aetherflow window
+		const prevWindow = event.action === this.data.actions.AETHERFLOW.id ? this.prevAetherflowWindow : this.prevDissipationWindow
+		const cooldown = this.data.getAction(event.action)?.cooldown ?? this.data.actions.AETHERFLOW.cooldown
+		let drift = 0
+		let cummulativeDrift = 0
+		if (this.aetherflowWindows.length > 1 && prevWindow) {
+			drift = event.timestamp - prevWindow.timestamp - cooldown
+			cummulativeDrift = prevWindow.cummulativeDrift + drift
+		}
```

Comment:
Should this calculation take into account Unable To Act downtime (e.g. where the player is incapacitated by a fight transition)?  The downtime module has a way to look up specifically windows where the player can't do anything (as opposed to just downtime where no enemies are targetable) so we don't count that drift time against the player.

Also, you could simplify your object tracking by simply doing all of the drift calculations at the end during the onComplete method, rather than needing to keep a hold of the last Aetherflow / Dissipation window as you go and then muddy that window information with a cumulative drift property that isn't really a part of just that specific window.

Reply:
Agreed on both counts.

To address your concern about what to do about wasted stacks around transition windows: A scholar needs a maximum of 10s (if they have no spell speed) to cast Aetherflow and spend all 3 stacks. Since Aetherflow's cooldown is 60s you have at least 50s of wiggle room to figure out when you want to cast your stacks. So any scripted downtime would have to last more than 50s to force the scholar into the situation where they have aetherflow stacks left and Aetherflow is off cooldown. The longest scripted downtime I saw was 20~25s, which leaves the scholar with 25~30s of wiggle room to figure out when to use their aetherflow stacks if it's going to come off cooldown during the scripted downtime. That gives the scholar at least 10~12 weave windows to spend their atherflow charges. To me that seems like more than enough time to spend aetherflow stacks to not be caught in a situation where you need to pick between drift and wasting stacks.

Reply:
That seems fair. Something else I noticed too was that scholar can cast Aetherflow without a target. So it depends on if the downtime is a mini cutscene or if the boss is just untargetable.  So I'll crunch some more numbers and figure out a good minimum UTA downtime duration where any UTA downtime starts forgiving drifting the cooldown. Probably something around 50-60s where it wouldn't be unreasonable that a scholar not drifting their cooldowns might end up caught with extra stacks of aetherflow during a transition.

Reply:
So upon further analysis I came up with 12 different edge cases based on whether or not the downtime was UTA vs invuln and a bunch of different timings. But the logic to handle all of that would have been, in my opinion, rather cumbersome and hard to test and maintain. So I opted for a simple check that moves the desired cast time to the end of a UTA window and gives the scholar a bit of buffer to reuse any remaining aetherflow stacks.

I used this log to check that I was bumping up the expected cast time correctly: https://www.fflogs.com/reports/1mBNCMdhpJ3kALgX#fight=29&type=casts&source=3